### ISSUE TITLE
Error making: python-pre-commit: dependencies not installed first

### ISSUE THREAD

**Issue by blueyed** [Jguer/yay#659]
#### Affected Version

yay v8.998.r83.g31aca68 - libalpm v11.0.1

#### Issue

It seems like dependencies are not installed before the package itself.

#### Steps to reproduce

```
% yay -S python-pre-commit                                                                                                                                                           🐍3.6.6
:: Checking for conflicts...
:: Checking for inner conflicts...
[Aur: 5]  python-cfgv-1.1.0-1  python-pre-commit-1.10.5-1  python-nodeenv-1.2.0-1  python-aspy-yaml-1.1.1-1  python-identify-1.0.5-2

  5 python-cfgv                      (Build Files Exist)
  4 python-pre-commit                (Build Files Exist)
  3 python-nodeenv                   (Build Files Exist)
  2 python-aspy-yaml                 (Build Files Exist)
  1 python-identify                  (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> a
:: Deleting (1/5): /home/daniel/.cache/pacaur/python-cfgv
:: Deleting (2/5): /home/daniel/.cache/pacaur/python-pre-commit
:: Deleting (3/5): /home/daniel/.cache/pacaur/python-nodeenv
:: Deleting (4/5): /home/daniel/.cache/pacaur/python-aspy-yaml
:: Deleting (5/5): /home/daniel/.cache/pacaur/python-identify
:: Downloaded PKGBUILD (1/5): python-cfgv
:: Downloaded PKGBUILD (2/5): python-aspy-yaml
:: Downloaded PKGBUILD (3/5): python-pre-commit
:: Downloaded PKGBUILD (4/5): python-identify
:: Downloaded PKGBUILD (5/5): python-nodeenv
  5 python-cfgv                      (Build Files Exist)
  4 python-pre-commit                (Build Files Exist)
  3 python-nodeenv                   (Build Files Exist)
  2 python-aspy-yaml                 (Build Files Exist)
  1 python-identify                  (Build Files Exist)
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> Diffs to show?
==> All

…

==> Proceed with install? [Y/n] 
:: Parsing SRCINFO (1/5): python-cfgv
:: Parsing SRCINFO (2/5): python-pre-commit
:: Parsing SRCINFO (3/5): python-nodeenv
:: Parsing SRCINFO (4/5): python-aspy-yaml
:: Parsing SRCINFO (5/5): python-identify
==> Making package: python-cfgv 1.1.0-1 (Sat 25 Aug 2018 11:02:44 PM CEST)
==> Retrieving sources...
  -> Downloading cfgv-1.1.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0   1270      0 --:--:-- --:--:-- --:--:--  1270
100   268  100   268    0     0    511      0 --:--:-- --:--:-- --:--:--  1558
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  6410  100  6410    0     0   7384      0 --:--:-- --:--:-- --:--:--  7384
==> Validating source files with md5sums...
    cfgv-1.1.0.tar.gz ... Passed
==> Cleaning up...
==> Making package: python-pre-commit 1.10.5-1 (Sat 25 Aug 2018 11:02:46 PM CEST)
==> Retrieving sources...
  -> Downloading pre_commit-1.10.5.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0   1140      0 --:--:-- --:--:-- --:--:--  1150
100   281  100   281    0     0    892      0 --:--:-- --:--:-- --:--:--   892
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  125k  100  125k    0     0   245k      0 --:--:-- --:--:-- --:--:--  245k
==> Validating source files with md5sums...
    pre_commit-1.10.5.tar.gz ... Passed
==> Cleaning up...
==> Making package: python-nodeenv 1.2.0-1 (Sat 25 Aug 2018 11:02:47 PM CEST)
==> Retrieving sources...
  -> Downloading nodeenv-1.2.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0    991      0 --:--:-- --:--:-- --:--:--   991
100   324  100   324    0     0    981      0 --:--:-- --:--:-- --:--:--   981
100 32844  100 32844    0     0  71400      0 --:--:-- --:--:-- --:--:-- 71400
==> Validating source files with sha256sums...
    nodeenv-1.2.0.tar.gz ... Passed
==> Cleaning up...
==> Making package: python-aspy-yaml 1.1.1-1 (Sat 25 Aug 2018 11:02:48 PM CEST)
==> Retrieving sources...
  -> Downloading aspy.yaml-1.1.1.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0   1232      0 --:--:-- --:--:-- --:--:--  1232
100   278  100   278    0     0    888      0 --:--:-- --:--:-- --:--:--   888
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  2026  100  2026    0     0   4502      0 --:--:-- --:--:-- --:--:--  4502
==> Validating source files with md5sums...
    aspy.yaml-1.1.1.tar.gz ... Passed
==> Cleaning up...
==> Making package: python-identify 1.0.5-2 (Sat 25 Aug 2018 11:02:49 PM CEST)
==> Retrieving sources...
  -> Downloading v1.0.5.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   127    0   127    0     0    153      0 --:--:-- --:--:-- --:--:--   153
100  8205    0  8205    0     0   4863      0 --:--:--  0:00:01 --:--:-- 1335k
==> Validating source files with sha256sums...
    v1.0.5.tar.gz ... Passed
==> Cleaning up...
==> Making package: python-cfgv 1.1.0-1 (Sat 25 Aug 2018 11:02:51 PM CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found cfgv-1.1.0.tar.gz
==> Validating source files with md5sums...
    cfgv-1.1.0.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting cfgv-1.1.0.tar.gz with bsdtar
==> Starting prepare()...
==> Sources are ready.
==> Making package: python-cfgv 1.1.0-1 (Sat 25 Aug 2018 11:02:53 PM CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting build()...
running build
running build_py
creating build
creating build/lib
copying cfgv.py -> build/lib
running build
running build_py
creating build
creating build/lib
copying cfgv.py -> build/lib
==> Entering fakeroot environment...
==> Starting package_python-cfgv()...
running install
running install_lib
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr/lib
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr/lib/python3.6
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr/lib/python3.6/site-packages
copying build/lib/cfgv.py -> /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr/lib/python3.6/site-packages
byte-compiling /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr/lib/python3.6/site-packages/cfgv.py to cfgv.cpython-36.pyc
writing byte-compilation script '/tmp/tmp9dxkm12p.py'
/home/daniel/.pyenv/versions/3.6.6/bin/python /tmp/tmp9dxkm12p.py
removing /tmp/tmp9dxkm12p.py
running install_egg_info
running egg_info
writing cfgv.egg-info/PKG-INFO
writing dependency_links to cfgv.egg-info/dependency_links.txt
writing requirements to cfgv.egg-info/requires.txt
writing top-level names to cfgv.egg-info/top_level.txt
reading manifest file 'cfgv.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'cfgv.egg-info/SOURCES.txt'
Copying cfgv.egg-info to /home/daniel/.cache/pacaur/python-cfgv/pkg/python-cfgv/usr/lib/python3.6/site-packages/cfgv-1.1.0-py3.6.egg-info
running install_scripts
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "python-cfgv"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Starting package_python2-cfgv()...
running install
running install_lib
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr/lib
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr/lib/python2.7
creating /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr/lib/python2.7/site-packages
copying build/lib/cfgv.py -> /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr/lib/python2.7/site-packages
byte-compiling /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr/lib/python2.7/site-packages/cfgv.py to cfgv.pyc
writing byte-compilation script '/tmp/tmpVPGU3D.py'
/usr/bin/python2 -O /tmp/tmpVPGU3D.py
removing /tmp/tmpVPGU3D.py
running install_egg_info
running egg_info
writing requirements to cfgv.egg-info/requires.txt
writing cfgv.egg-info/PKG-INFO
writing top-level names to cfgv.egg-info/top_level.txt
writing dependency_links to cfgv.egg-info/dependency_links.txt
reading manifest file 'cfgv.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'cfgv.egg-info/SOURCES.txt'
Copying cfgv.egg-info to /home/daniel/.cache/pacaur/python-cfgv/pkg/python2-cfgv/usr/lib/python2.7/site-packages/cfgv-1.1.0-py2.7.egg-info
running install_scripts
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "python2-cfgv"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: python-cfgv 1.1.0-1 (Sat 25 Aug 2018 11:02:56 PM CEST)
==> Cleaning up...
loading packages...
resolving dependencies...
looking for conflicting packages...

Package (1)  New Version  Net Change

python-cfgv  1.1.0-1        0.06 MiB

Total Installed Size:  0.06 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                                                                     [--------------------------------------------------------------------] 100%
(1/1) checking package integrity                                                                                   [--------------------------------------------------------------------] 100%
(1/1) loading package files                                                                                        [--------------------------------------------------------------------] 100%
(1/1) checking for file conflicts                                                                                  [--------------------------------------------------------------------] 100%
(1/1) checking available disk space                                                                                [--------------------------------------------------------------------] 100%
:: Processing package changes...
(1/1) installing python-cfgv                                                                                       [--------------------------------------------------------------------] 100%
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
==> Making package: python-pre-commit 1.10.5-1 (Sat 25 Aug 2018 11:02:57 PM CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Missing dependencies:
  -> python-nodeenv
  -> python-aspy-yaml
  -> python-identify
==> ERROR: Could not resolve all dependencies.
Error making: python-pre-commit
```

**Comment by Morganamilo** [comment]
Strange, that's the first time I've seen that happen. Installed fine on my end though. Do you still get an error the second time round if you run it again?

**Comment by blueyed** [comment]
Yes.
But I get less errors depending on if I clean-build or not (or something like that).

It also ended up being installed once when I've tried different answers etc - but in the end I've just added it to a virtualenv manually.

```
% cat ~/.config/yay/config.json
{
	"buildDir": "/home/daniel/.cache/pacaur/",
	"editor": "",
	"editorflags": "",
	"makepkgbin": "/usr/bin/makepkg",
	"makepkgconf": "",
	"pacmanbin": "/usr/bin/pacman",
	"pacmanconf": "/etc/pacman.conf",
	"tarbin": "/usr/bin/bsdtar",
	"redownload": "no",
	"rebuild": "no",
	"answerclean": "",
	"answerdiff": "All",
	"answeredit": "",
	"answerupgrade": "",
	"gitbin": "git",
	"gpgbin": "gpg",
	"gpgflags": "",
	"mflags": "",
	"sortby": "votes",
	"gitflags": "",
	"removemake": "ask",
	"requestsplitn": 50,
	"sortmode": 0,
	"completionrefreshtime": 7,
	"sudoloop": false,
	"timeupdate": false,
	"devel": true,
	"cleanAfter": false,
	"gitclone": true,
	"provides": true,
	"pgpfetch": true,
	"upgrademenu": true,
	"cleanmenu": true,
	"diffmenu": true,
	"editmenu": false,
	"combinedupgrade": false,
	"useask": false
}
```

**Comment by Morganamilo** [comment]
Maybe specifying the order might have worded as a work around
```
yay -S python-nodeenv python-aspy-yaml python-identify python-cfgv  python-pre-commit
```

Or at the very least it probably would have worked if you tried it one command at a time.

I'm still really confused though. Dependencies are resolved are akin to post order transferal. The package added to the command line should always be the last to be installed.

And even when I require the exact same dependencies as you I still can't reproduce it
```
:: Checking for conflicts...
:: Checking for inner conflicts...
[Aur: 5]  python-nodeenv-1.2.0-1  python-aspy-yaml-1.1.1-1  python-identify-1.0.5-2  python-cfgv-1.1.0-1  python-pre-commit-1.10.5-1

  5 python-nodeenv                   (Build Files Exist)
  4 python-aspy-yaml                 (Build Files Exist)
  3 python-identify                  (Build Files Exist)
  2 python-cfgv                      (Build Files Exist)
  1 python-pre-commit                (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==>  
```

**Comment by blueyed** [comment]
Just tried it again and it worked.

yay v8.998.r89.g73a171f - libalpm v11.0.1

It does not look like anything from 31aca68..73a171f would have fixed this, except for d0cb79d maybe?!

Closing for now anyway.
